### ISSUE TITLE
Distro Packaging

### ISSUE THREAD

**Issue by hrehfeld** [syncthing-gtk/syncthing-gtk#12]
I'd prefer to have distro packaging in separate repos that are hosted by this (or the renamed) organization.

For Archlinux, as long as it's in the user package repo, there's a separate repository anyways: https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=syncthing-gtk https://aur.archlinux.org/packages/syncthing-gtk In my experience, a separate repo is basically no extra work once you get used to it (archlinux user).

Also, I found the package build files slightly confusing in the project history.

**Comment by Salamandar** [comment]
Agreed with all of the above. Although, i'd prefer keeping the AUR package hosted only at one place : i'll forget to push here one day…

**Comment by Rudd-O** [comment]
My PR [here](https://github.com/Rudd-O/syncthing-gtk/blob/master/syncthing-gtk.spec) had RPM packaging.

What's the benefit of a separate repository?  It's much easier to build packages if the user only has to download one repo.

**Comment by Salamandar** [comment]
> What's the benefit of a separate repository?

If you add Debian and Fedora packaging, why not all the existing distros out there ?
Also a lot of distros maintain the build descriptions in their repo (at least Arch, Gentoo, Homebrew, Nix, etc. That's for example why i dont want to push the AUR packages (that i maintain) here : they will someday become unmaintained/out of date compared to the aur repos.

> It's much easier to build packages if the user only has to download one repo.

Users rarely build distro packages themselves. Packagers do. I'm ok for hosting here some package descriptions that aren't maintained elsewhere (like windows NSIS files). But for linux distros that have repositories for that, I am not.

**Comment by Rudd-O** [comment]
> If you add Debian and Fedora packaging, why not all the existing distros out there ?

**Exactly!  Why not?**

The ZFS people carry packaging for like 12 different distros right in the source repo.

If 12 different people from 12 different distros want to contribute packaging scripts, we should accept them!

> Also a lot of distros maintain the build descriptions in their repo (at least Arch, Gentoo, Homebrew, Nix, etc. That's for example why i dont want to push the AUR packages (that i maintain) here : they will someday become unmaintained/out of date compared to the aur repos.

Sure!  If some distributors prefer to maintain their packaging recipes outside this repo, that's more than welcome.

*What I'm saying is that we should not prohibit people from contributing packaging recipes directly into this repo.*  Nothing more, nothing less.

> Users rarely build distro packages themselves. Packagers do. 

Yes, *and?*  How does carrying packaging scripts in this repo cause a problem for anyone?

*Right this very instant* my fork of this repo has everything I need to do the following and get a package (which my CI/CD does automatically, because it's already in the repo):

```
    make rpm
    # bam!  packages!
```

How cool is that?  I don't have to wait for Fedora to package this project — I can push button get package.

**Comment by Salamandar** [comment]
> _What I'm saying is that we should not prohibit people from contributing packaging recipes directly into this repo._ Nothing more, nothing less.

Hmmm… alright, I get your point. Although I don't quite understand what the use case might be, I won't oppose it.

What is bugging me is that usually, package descriptions *pull* the code from upstream, instead of *being* upstream. So, for example, I'm not sure how to write a archlinux PKGBUILD that uses the code 1 upper level in the filesystem. Might not be too difficult though.

Please do your merge request, we'll discuss the details there.

**Comment by Salamandar** [comment]
@Rudd-O ping :)

**Comment by Rudd-O** [comment]
I'm sorry, I've been hella busy.  I will figure out what to do next week.

**Comment by Salamandar** [comment]
FYI I just merged the flatpak packaging. It contains Syncthing too.

**Comment by UnderEu** [comment]
+1 in the need for a Ubuntu package of this one, even better if comes with its own repo or PPA

**Comment by cromefire** [comment]
Just some packaging that's not sandboxed would be awesome (or a fix for the flatpak). The flatpak is great and all, but I don't want to rebuild my whole syncthing config and I can't make it connect to local syncthing (even when I give it access via flat seal it doesn't find the binary).

**Comment by tehbuhtig** [comment]
@cromefire - I also failed to get the flatpak to work w/ external syncthing. after a while I gave up and just used the syncthing provided w/ the flatpak. I found I didn't have to rebuild my config if I copied the syncthing configuration files from $HOME/.config/syncthing to $HOME/.var/app/me.kozec.syncthingtk/config/syncthing.
